src/main/kotlin/org/rust/ide/docs/RsDocumentationProvider.kt

```diff
@@ -17,89 +17,159 @@ class RsDocumentationProvider : AbstractDocumentationProvider() {
 
         val name = if (element is RsMod) element.modName else element.name
         val header = if (name != null) "<pre>$name</pre>\n" else ""
-        val functionSignature = (element as? RsFunction)?.formatSignature()
+        val functionSignature = (element as? RsFunction)?.signatureText
         val signature = if (functionSignature != null) "<pre>$functionSignature</pre>\n" else ""
         val doc = element.documentationAsHtml() ?: ""
         return header + signature + doc
     }
 
-    override fun getQuickNavigateInfo(element: PsiElement, originalElement: PsiElement?) = when (element) {
-        is RsPatBinding -> getQuickNavigateInfo(element)
-        is RsNamedElement -> element.getQuickNavigateInfo()
-        else -> null
+    override fun getQuickNavigateInfo(element: PsiElement, originalElement: PsiElement?): String? {
+        val formatString = when (element) {
+            is RsFunction -> element.signatureText
+            is RsStructItem -> element.signatureText
+            is RsFieldDecl -> element.signatureText
+            is RsEnumItem -> element.signatureText
+            is RsEnumVariant -> element.signatureText
+            is RsTraitItem -> element.signatureText
+            is RsTypeAlias -> element.signatureText
+            is RsConstant -> element.signatureText
+            is RsSelfParameter -> element.signatureText
+            is RsTypeParameter -> element.signatureText
+            is RsLifetimeDecl -> element.signatureText
```

Comment:
The ton of short `signatureText` functions seems a bit repetitive. Perhaps it's better to inline them into this `when`? 

Also, some of `signatureText` are `fun`s and some are `val`s